creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -377,37 +388,73 @@ class SnapbackSM {
    * }
    */
   async determineNewReplicaSet ({ primary, secondary1, secondary2, wallet, unhealthyReplicasSet, healthyNodes }) {
-    let currentNodeHasExistingUserState = true
+    let newReplicaNodesSet = new Set()
     let i = 0
-    let newSecondary
-    while (currentNodeHasExistingUserState && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
-      // Make sure current replica set will not be chosen as potential nodes for the new replica set
-      const healthyPotentialPeers = healthyNodes.filter(node => node !== primary && node !== secondary1 && node !== secondary2)
-
+    let issueReconfig = false
+    let newPrimary, newSecondary1, newSecondary2
+    while (newReplicaNodesSet.size !== NUMBER_OF_REPLICA_SET_NODES && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
       // Select a random node that is not from the current replica set
-      newSecondary = healthyPotentialPeers[Math.floor(Math.random() * healthyPotentialPeers.length)]
+      let randomHealthyNode = healthyNodes[Math.floor(Math.random() * healthyNodes.length)]
+
+      // If node is already present in set, keep finding a unique healthy node
+      if (newReplicaNodesSet.has(randomHealthyNode)) continue
 
       // Check to make sure that the newly selected secondary does not have existing user state
       try {
-        const clockValue = await CreatorNode.getClockValue(newSecondary, wallet)
-        if (clockValue === -1) currentNodeHasExistingUserState = false
+        const clockValue = await CreatorNode.getClockValue(randomHealthyNode, wallet)
+        if (clockValue === -1) {
+          newReplicaNodesSet.add(randomHealthyNode)
+        }
       } catch (e) {
         // Something went wrong in checking clock value. Reselect another secondary.
-        this.logError(e.message)
+        this.logError(`[determineNewReplicaSet] ${e.message}`)
       }
     }
 
-    if (currentNodeHasExistingUserState) throw new Error(`Unable to select new secondary given current healthy Content Nodes=[${healthyNodes}]`)
-
-    const currentHealthySecondary = unhealthyReplicasSet.has(secondary1) ? secondary2 : secondary1
+    let newReplicaNodes = Array.from(newReplicaNodesSet)
+    // We need a minimum of `NUMBER_OF_REPLICA_SET_NODES` to perform the reconfig op. In the case that
+    // the max number replica set attempts were made, push null
+    while (newReplicaNodes.length < NUMBER_OF_REPLICA_SET_NODES) {
```

Comment:
what is this push null for?

Reply:
in the case that the max number of attempts were reached and there are not enough nodes in `newReplicaNodes`, push null values. this ensures that the length of `newReplicaNodes` will always be what we expect it to be, and there will be no array out of bounds access error in the if/else case a few lines down

Reply:
will update with throwing error + it will be caught in caller method